### PR TITLE
161400356  implement c2 feedback 

### PR DESCRIPTION
**Pull request by charitymarani**
**What does this PR do?**
This PR implements all challenge 2 feedback

**Description of Task to be completed:**
1. Validated registration with existing email
2. Written error handlers to catch errors
3. Implemented the use of lists of dictionaries instead of a dictionary of dictionaries
4. Written tests for the error handlers

**How to manually test**

`$ pip install virtualenv`
NB This install virtualenv Globally(virtual environment allows you to create independent envs isolated from your actual physical machine)
`$ virtualenv -p python3 myenv`
`$ source myenv/bin/activate`
Clone this repository: `$ git clone https://github.com/charitymarani/store_manager_api.git`
navigate into the repository after cloning
`$ git checkout to ch-implement-c2-feedback-161400356`  to switch from master to this branch
`$ pip install -r requirements.txt`
Launch postman and test with any endpoint for instance:

`POST /api/v1/auth/register`
headers = {content_type:application/json}

**Any background information**
None
**Relevant story**
[#161400356](https://www.pivotaltracker.com/story/show/161400356)

### CODE REVIEW

File: application/__init__.py [hunk 1]
```diff
@@ -38,9 +39,33 @@ def check_if_token_blacklist(decrypted_token):
         json_token_identifier = decrypted_token['jti']
         return json_token_identifier in BLACKLIST
 
+    '''Error handlers'''
```
Comment (Bonifase) [inline application/__init__.py:41]: Nice work. Consider putting all error handlers in one separate file in challenge 3

File: application/api/v1/models/auth_models.py [hunk 2]
```diff
@@ -1,40 +1,50 @@
 import string
 from flask import jsonify
 from werkzeug.security import generate_password_hash, check_password_hash
-from ..utils import get_item_by_key
+from ..utils import get_item_by_key, get_all
+
+USERS_DICT = []
```
Comment (Bonifase) [inline application/api/v1/models/auth_models.py]: Is this a dict or an list? Is this a dict or list? I suppose it should be a list

File: application/api/v1/models/product_model.py [hunk 3]
```diff
@@ -1,29 +1,31 @@
-from ..utils import get_item_by_key
+from ..utils import get_item_by_key, get_all
+
+PRODUCTS_DICT = []
```
Comment (Bonifase) [inline application/api/v1/models/product_model.py]: Is this a dict or list? I suppose it should be a list

**Comment by Nicanor008** [comment]
Nice work, I like how you have organized your work.

File: application/api/v1/models/product_model.py (hunk 3 above)
Comment (charitymarani) [reply application/api/v1/models/product_model.py]: It is a list, I didn't change the variable names when I shifted from using dictionaries but I'll consider that

File: application/__init__.py (hunk 1 above)
Comment (charitymarani) [reply application/__init__.py:41]: Got it!

**Pull request merged**